catalog/core/catalog-core-api/src/main/java/ddf/catalog/plugin/PrePolicyPlugin.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin;
+
+import java.util.Map;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+
+/**
+ * The {@link PrePolicyPlugin} extension point allows for request/response processing before any
+ * policies are implemented at the very beginning of the plugin chain.
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public interface PrePolicyPlugin {
```

Comment:
Having a class named `PrePolicyPlugin` that offers methods like `processPreDelete` and `processPostDelete` is going to lead to a little bit of confusion. You might consider taking advantage of the fact there's a second Latin-based prefix you can use here to indicate your plugin should operate prior to the Policy plugins and rename this class `AntePolicyPlugin`.

It's a trifling point and others may dislike the suggested name, but calling pre+pre and pre+post methods is awfully ugly.